src/clr-angular/forms/select/select-container.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ContentChild, OnDestroy, Optional } from '@angular/core';
+import { SelectMultipleControlValueAccessor } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
+import { ControlIdService } from '../common/providers/control-id.service';
+import { ClrLabel } from '../common/label';
+import { ControlClassService } from '../common/providers/control-class.service';
+
+@Component({
+  selector: 'clr-select-container',
+  template: `    
+        <ng-content select="label"></ng-content>
+        <label *ngIf="!label && addGrid()"></label>
+        <div class="clr-control-container" [ngClass]="controlClass()">
+            <div [class.clr-multiselect-wrapper]="multi" [class.clr-select-wrapper]="!multi">
+                <ng-content select="[clrSelect]"></ng-content>
+                <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+            </div>
+            <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+            <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+        </div>
+    `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [IfErrorService, NgControlService, ControlIdService, ControlClassService],
+})
+export class ClrSelectContainer implements DynamicWrapper, OnDestroy {
+  subscriptions: Subscription[] = [];
+  invalid = false;
+  _dynamic = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
+  @ContentChild(SelectMultipleControlValueAccessor) multiple: SelectMultipleControlValueAccessor;
+  multi = false;
+
+  constructor(
+    private ifErrorService: IfErrorService,
+    @Optional() private layoutService: LayoutService,
+    private controlClassService: ControlClassService,
+    ngControlService: NgControlService
+  ) {
+    this.subscriptions.push(
+      this.ifErrorService.statusChanges.subscribe(control => {
+        this.invalid = control.invalid;
+      })
+    );
+    this.subscriptions.push(
+      ngControlService.controlChanges.subscribe(control => {
+        this.multi = control.valueAccessor instanceof SelectMultipleControlValueAccessor;
```

Comment:
I initially ticked at the fact that `multi`, which is used only for styling, is tied to Angular forms. But then I remembered we actually don't support our forms components to be used without Angular forms, so this is fine.

I don't know, I just felt like posting this. It felt weird to me that styling would break without Angular forms. But no change needed here.

Reply:
The alternative was to inspect the element and check for the multi property. I thought this was a little more elegant and localized.